### PR TITLE
8323023: GenShen: Region logging test fails intermittently

### PR DESCRIPTION
**Pull request by earthling-amzn**
This test used to exercise a specific Shenandoah feature, but we've long since rewritten that feature to be based on unified logging. The test doesn't need to retest unified logging features.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JDK-8323023](https://bugs.openjdk.org/browse/JDK-8323023): GenShen: Region logging test fails intermittently (**Bug** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - Committer)
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer) ⚠️ Review applies to [e28aca2b](https://git.openjdk.org/shenandoah/pull/380/files/e28aca2be45662f1a8583e00127943e48d1bf675)
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah.git pull/380/head:pull/380` \
`$ git checkout pull/380`

Update a local copy of the PR: \
`$ git checkout pull/380` \
`$ git pull https://git.openjdk.org/shenandoah.git pull/380/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 380`

View PR using the GUI difftool: \
`$ git pr show -t 380`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/380.diff">https://git.openjdk.org/shenandoah/pull/380.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/shenandoah/pull/380#issuecomment-1877959354)

### CODE REVIEW

File: test/hotspot/jtreg/gc/shenandoah/TestShenandoahRegionLogging.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+public class TestShenandoahRegionLogging {
+    private static final long TARGET_MB = Long.getLong("target", 1);
+    private static Object sink;
+
+    public static void main(String[] args) throws Exception {
+        long count = TARGET_MB * 1024 * 1024 / 16;
+        for (long c = 0; c < count; c++) {
+            sink = new Object();
+            Thread.sleep(1);
```
Comment (shipilev) [inline test/hotspot/jtreg/gc/shenandoah/TestShenandoahRegionLogging.java:47]: Pre-existing, but: The problem with doing this sleep is that we are doing about 64K of 1ms sleeps in this tests. Even if we manage to do 1ms sleep, that is already 65 seconds spent in this test. But we might as well spend single-digit ms per sleep. For 10ms sleeps, that is 10 minutes for a test, which would be playing it close to timeout.

So I wonder if this test should be time-bound.

**Review by shipilev — APPROVED** [review]
Looks okay, but there is a pre-existing issue, I think.

**Review by kdnilsen — APPROVED** [review]

File: test/hotspot/jtreg/gc/shenandoah/TestShenandoahRegionLogging.java (hunk 1 above)
Comment (kdnilsen) [reply test/hotspot/jtreg/gc/shenandoah/TestShenandoahRegionLogging.java:47]: Would it be better to reduce the frequency of the sleep, to one in 10, for example?

Comment (earthling-amzn) [reply test/hotspot/jtreg/gc/shenandoah/TestShenandoahRegionLogging.java:47]: The test is really just waiting for region sampling to write to the logs. I think one call to `System.gc` would suffice. I'll see if I can reduce the runtime for this test.

**Review by shipilev — APPROVED** [review]
Looks good, as long the test runs well :) Might want to try with JTREG=REPEAT_COUNT=100 or something.

**Review by ysramakrishna — APPROVED** [review]
Looks good. Should the test additionally check the contents of one of the files to find a line describing a region snapshot?

**Pull request closed without merging**